### PR TITLE
Suggest either addresses or locations; Moves addressfinder functions to their own file

### LINKED ISSUE #67: Re-instate some street address functionality
As a users i want to be able to search by location, but also exact street addresses, do that i can refer to exact locations rather than wide areas.

- [ ] search by street address
- [ ] Radius options ( 10, 25, 50, more) km. probably radio button
- [ ] defaults to 25km radius (if you have entered a street address)

### PR DESCRIPTION
**Pull request by lamorrison**
Provide suggestions from either the location or address api as appropriate

Also allows a blank location to remove any existing location parameters in the URL

@mischa-s has added the radius selector and search integration

close #67

### CODE REVIEW

File: src/utilities/addressfinder.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+  // Get teh full record for a location specified by its primay key (pxid)
+  const getLocationMetadata = (pxid) => {
+    const query = queryString.stringify({
+      key: ADDRESS_FINDER_KEY,
+      format:'json',
+      pxid: pxid
+    })
+    
+    const url = `https://api.addressfinder.io/api/nz/location/info?${query}`
```
Comment (Br3nda) [inline src/utilities/addressfinder.js]: `api.addressfinder.io` could be kept in .env instead.

File: src/utilities/addressfinder.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      strict: 2
+    })
+
+    const url = `https://api.addressfinder.io/api/nz/address?${query}`
+    
+    return axios.get(url)
+      .then(response => {
+        return response.data.completions
+      });
+  };
```
Comment (Br3nda) [inline src/utilities/addressfinder.js]: could DRY like this:

```
addressFinderQuery(method, query) {
  query['key'] = ADDRESS_FINDER_KEY;
  query['format'] = 'json';
   const url = `${ADDRESSFINDER_BASEURL}${method}?${query}`

    return axios.get(url)
      .then(response => {
        return response.data.completions
      });
}
```

**Comment by mischa-s** [comment]
Looking at #67 radius is included in the checklist. Do you want to add it to this PR @lamorrison  or should we make a separate ticket for it?
"
- [ ] search by street address
- [ ] Radius options ( 10, 25, 50, more) km. probably radio button
- [ ] defaults to 25km radius (if you have entered a street address)
"

**Comment by Br3nda** [comment]
This pull request **introduces 3 alerts** and **fixes 2** when merging 70d826332949c5a9813d7fb2cebe59ecbc143c8b into 0ec6bb5ef72608872596f92fe8c3f2f9998435fd - [view on LGTM.com](https://lgtm.com/projects/g/ServiceInnovationLab/FSD/rev/pr-5c60098106d70003d9bafb3de767f2cf74316ce6)

**new alerts:**

* 2 for Missing variable declaration
* 1 for Unused or undefined state property

**fixed alerts:**

* 2 for Missing variable declaration

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by lamorrison** [comment]
> Can we have some tests please

New tests:

  Scenario: Search controls
    ...
    And The radius selector should be set to "25" km radius  

Scenario: I search near Thorndon Quay and get some results
    Given I enter "Thorndon Quay" into the "address-autosuggest" input
    And I click on the first address suggestion
    Then I see a list of service providers

  Scenario: Address searches will autosuggest regions
      Given I enter "Thorndon Quay" into the "address-autosuggest" input
      Then The first suggestion should be "Thorndon Quay, Pipitea, Wellington"

  Scenario: Address searches will autosuggest street addresses
      Given I enter "191 Thorndon" into the "address-autosuggest" input
      Then The first suggestion should be "191 Thorndon Quay, Pipitea, Wellington"

**Comment by Br3nda** [comment]
This pull request **introduces 1 alert** and **fixes 3** when merging ba8d2a1f0736d16544d9af813a06e540089ea4f1 into 0ec6bb5ef72608872596f92fe8c3f2f9998435fd - [view on LGTM.com](https://lgtm.com/projects/g/ServiceInnovationLab/FSD/rev/pr-27b47b72493ebc8e0e22d0aa9b010fe4b4d2fe2d)

**new alerts:**

* 1 for Unused or undefined state property

**fixed alerts:**

* 3 for Missing variable declaration

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by Br3nda** [comment]
This pull request **introduces 1 alert** and **fixes 3** when merging 6dcd484dd07fa6647133b9c15609161fd9d3a692 into 227199b12f3f3459df4d17396ce4822dcdfcbc38 - [view on LGTM.com](https://lgtm.com/projects/g/ServiceInnovationLab/FSD/rev/pr-844cfb1ca9c4179b150588d2e9f7cc56be0ee909)

**new alerts:**

* 1 for Unused or undefined state property

**fixed alerts:**

* 3 for Missing variable declaration

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by Br3nda** [comment]
This pull request **introduces 1 alert** and **fixes 3** when merging 5574efff890d8e06f6e07f55b65580921cd9c135 into 227199b12f3f3459df4d17396ce4822dcdfcbc38 - [view on LGTM.com](https://lgtm.com/projects/g/ServiceInnovationLab/FSD/rev/pr-767a73418c72f87f72acecb1fc022e5c0bb67535)

**new alerts:**

* 1 for Unused or undefined state property

**fixed alerts:**

* 3 for Missing variable declaration

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**